packages/instrumentation-node-cache/src/node-cache.ts

```diff
@@ -0,0 +1,132 @@
+            'node-cache',
+            ['>=5.0.0'],
+            this.patch.bind(this)
+        );
+        return module;
+    }
+
+    private patch(moduleExports: NodeCacheType, moduleVersion: string) {
+        const self = this;
+        const origConstructor = moduleExports as unknown as Function;
+        const origPrototype = moduleExports.prototype;
+
+        function PatchedNodeCache(options = {}) {
+            const inst: NodeCache = new (origConstructor as any)(options);
+            ['get', 'take', 'del', 'getTtl', 'has', 'set', 'mget', 'flushAll'].forEach((op) =>
+                self._wrap(
+                    inst,
+                    op as keyof NodeCache,
+                    self
+                        .patchClassFunction(
+                            op,
+                            (args: any[]) => `${op} ${args[0] ? args[0] : ''}`.trim(),
+                            moduleVersion
+                        )
+                        .bind(self)
+                )
+            );
+            self._wrap(
+                inst,
+                'mset',
+                self
+                    .patchClassFunction(
+                        'mset',
+                        (args: any[]) => `mset ${args[0].map((entry: NodeCache.ValueSetItem) => entry.key).join(',')}`,
+                        moduleVersion
+                    )
+                    .bind(self)
+            );
+            self._wrap(
+                inst,
+                'ttl',
+                self.patchClassFunction('ttl', (args: any[]) => `ttl ${args[0]} ${args[1]}`, moduleVersion).bind(self)
+            );
+            return inst;
+        }
+        PatchedNodeCache.prototype = origPrototype;
+        PatchedNodeCache.version = (moduleExports as any).version;
+        return PatchedNodeCache;
+    }
+
+    private patchClassFunction(opName: string, toStatement: (...args: any[]) => string, moduleVersion: string) {
+        const self = this;
+        return (originalFunc: Function) => {
+            return function func() {
+                if (self._config.requireParentSpan && trace.getSpan(context.active()) === undefined) {
+                    return originalFunc.apply(this, arguments);
+                }
+
+                const span = self.tracer.startSpan(`node-cache ${opName}`, {
+                    kind: SpanKind.INTERNAL,
+                    attributes: {
+                        [SemanticAttributes.DB_SYSTEM]: 'node-cache',
+                        [SemanticAttributes.DB_OPERATION]: opName,
+                        [SemanticAttributes.DB_STATEMENT]: toStatement(arguments),
+                    },
+                });
+
+                if (self._config.requestHook) {
+                    try {
+                        self._config.requestHook(span, {
+                            moduleVersion,
+                            operation: opName,
+                            args: Object.values(arguments),
```

Comment:
isn't arguments an array? why do we need to call `Object.values`?

Reply:
it's an array-like object (implementing IArguments interface).   
You can use it like array but typescript didn't approve this